### PR TITLE
Safelist the classes for common icon sizes

### LINKED ISSUE #451: Ensure icon size classes are not purged
## Problem

The icon sizes in `VIcon` component are set using dynamically-generated Tailwind classnames:
https://github.com/WordPress/openverse-frontend/blob/df42a228df54faa3f8fea4fc70fac5c50288564b/src/components/VIcon/VIcon.vue#L2-L8
In production, unused class names are purged, which might cause the icon size to not work correctly.

## Description

We should add the commonly-used icon class names to the `safelist` in tailwind config so that they are never purged.

## Alternatives

An alternative could  be to apply fully-qualified class names conditionally (ie `:class="size === 4 ? 'w-4 h-4' : 'w-5 h-5'"). But this would be more difficult to adjust if the sizes change, and it would be more difficult to accommodate more that 2 sizes.

## Implementation

- [x] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #451 by @obulat

## Description

This PR adds a `safelist` array to the tailwind config to ensure that the common icon size class names are not purged in production. 

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by dhruvkb — APPROVED** [review]
Please also add a comment referencing this file where the dynamic classes are actually used. Also note there that `4` and `5` are the only _safe_ values to use for size.

Approving to unblock merge.

**Comment by obulat** [comment]
Based on @panchovm 's comment, I'm going to add more sizes:

> In general, the sizes I’m using in Figma are 24px and 18px. This includes a safe area and follows the default size defined in the WP Design Library.
> Also, we have the play and pause sizes for the audio control button, and CC license icons in three sizes (14px, 16px, 26px)

File: tailwind.config.js [hunk 1]
```diff
@@ -1,5 +1,12 @@
 module.exports = {
-  purge: ['src/**/*.{vue,js,jsx,ts,tsx,mdx}', './nuxt.config.js'],
+  purge: {
+    content: ['src/**/*.{vue,js,jsx,ts,tsx,mdx}', './nuxt.config.js'],
+    // In production, dynamically-created class names are purged as unused:
+    // https://tailwindcss.com/docs/optimizing-for-production#writing-purgeable-html
+    // The commonly-used icon size classes are safe-listed here
+    // to ensure that they are not removed:
```
Comment (dhruvkb) [inline tailwind.config.js:7]: Minor nitpick about code style. Long multiline comment would be better written as `/* */`.

File: src/components/VIcon/VIcon.vue [hunk 2]
```diff
@@ -51,11 +51,15 @@ export default {
     },
     /**
      * the size of the icon in terms of the Tailwind height and width scale; The
-     * icon, being square, uses this for both dimensions.
+     * icon, being square, uses this for both dimensions. Possible values:
+     * 4 - 16px(1rem), 5 - 20px(1.25rem), 6 - 24px(1.5rem).
```
Comment (dhruvkb) [inline src/components/VIcon/VIcon.vue:55]: Let's not reference `px` values here. Keeping the numeric sizes decoupled from actual values seems better to me.

Comment (zackkrida) [reply src/components/VIcon/VIcon.vue:55]: Probably best not to mention specific values in the comment at all—typedoc or a link to the tailwind config would be ideal.

Comment (dhruvkb) [reply src/components/VIcon/VIcon.vue:55]: We should still clarify that the acceptable values are 4, 5 and 6 only.

**Review by zackkrida — APPROVED** [review]
Going to unblock this instead of requesting revision of a minor comment gripe.

**Pull request merged**